src/battle_ai_switch_items.c

```diff
@@ -208,7 +210,10 @@ static bool8 FindMonThatAbsorbsOpponentsMove(void)
             continue;
 
         monAbility = GetMonAbility(&party[i]);
-        if (absorbingTypeAbility == monAbility && Random() & 1)
+        if (absorbingTypeAbility == monAbility
+        || (absorbingTypeAbility == ABILITY_WATER_ABSORB && (monAbility == ABILITY_STORM_DRAIN || monAbility == ABILITY_DRY_SKIN))
+        || (absorbingTypeAbility == ABILITY_VOLT_ABSORB && (monAbility == ABILITY_MOTOR_DRIVE || monAbility == ABILITY_LIGHTNING_ROD))
```

Comment:
hmm, I think it might be better to have a small array that correlates each ability to the type it absorbs,

Reply:
Updated following this structure, almost exactly lifted from Buffel's Inclement Emerald so if this gets accepted please also credit them :)